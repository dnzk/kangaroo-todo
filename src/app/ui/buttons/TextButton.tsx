import style from "./buttons.module.scss"
import { ButtonHTMLAttributes, DetailedHTMLProps } from "react"

type ButtonProps = DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>

export default function TextButton(props: ButtonProps) {
  return (
    <button
      {...props}
      className={style['text-button']}
    >
      {props.children}
    </button>
  )
}
