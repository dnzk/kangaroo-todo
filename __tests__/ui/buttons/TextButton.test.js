import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import TextButton from '../../../src/app/ui/buttons/TextButton'

describe('TextButton', () => {
  it('renders correctly', () => {
    const { container } = render(<TextButton>text</TextButton>)
    expect(container).toMatchSnapshot()
  })
})
