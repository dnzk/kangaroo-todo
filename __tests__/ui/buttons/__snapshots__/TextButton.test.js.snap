// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton renders correctly 1`] = `
<div>
  <button
    class="text-button"
  >
    text
  </button>
</div>
`;
